src/validation/__tests__/DefaultValuesOfCorrectType-test.js

```diff
@@ -68,6 +68,44 @@ describe('Validate: Variable default values of correct type', () => {
     `);
   });
 
+  it('variables with valid default null values', () => {
+    expectPassesRule(DefaultValuesOfCorrectType, `
+      query WithDefaultValues(
+        $a: Int = null,
+        $b: String = null,
+        $c: ComplexInput = { requiredField: true, intField: null }
+      ) {
+        dog { name }
+      }
+    `);
+  });
+
+  it('variables with invalid default null values', () => {
+    expectFailsRule(DefaultValuesOfCorrectType, `
+      query WithDefaultValues(
+        $a: Int! = null,
+        $b: String! = null,
+        $c: ComplexInput = { requiredField: null, intField: null }
+      ) {
+        dog { name }
+      }
+    `, [
+      defaultForNonNullArg('a', 'Int!', 'Int', 3, 20),
+      badValue('a', 'Int!', 'null', 3, 20, [
```

Comment:
Interesting: I'm surprised to see two errors each for these cases, but I suppose it makes sense.